### PR TITLE
Update GeckoView and use loadData 🦎

### PR DESCRIPTION
**Pull request by ekager**

### CODE REVIEW

File: app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java [hunk 1]
```diff
@@ -269,26 +265,32 @@ public void onCanGoForward(GeckoSession session, boolean canGoForward) {
                 }
 
                 @Override
-                public boolean onLoadRequest(GeckoSession session, String uri, @TargetWindow int target) {
+                public void onLoadRequest(GeckoSession session, String uri, int target, GeckoSession.Response<Boolean> response) {
```
Comment (sputnikci) [inline app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java]: \[PMD\] INFO: The method 'onLoadRequest' has a Cyclomatic Complexity of 8\.

**Comment by pocmo** [comment]
@ekager Is this done and should I re-review this? :)

File: app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java [hunk 2]
```diff
@@ -197,17 +213,36 @@ public void fullScreenExited() {
...
+                    if (scheme == null || (!scheme.equals("http") && !scheme.equals("https"))) {
+                        // We are ignoring everything that is not http or https. This is a limitation of
+                        // Android's download manager. There's no reason to show a download dialog for
+                        // something we can't download anyways.
+                        Log.w(TAG, "Ignoring download from non http(s) URL: " + response.uri);
+                        return;
+                    }
+
+                    if (callback != null) {
+                        final Download download = new Download(response.uri, "Mozilla/5.0 (Android 8.1.0; Mobile; rv:60.0) Gecko/60.0 Firefox/60.0",
```
[the remark this answers is not in the document]
Comment (ekager) [reply app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java]: https://bugzilla.mozilla.org/show_bug.cgi?id=1454444

File: app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java [hunk 3]
```diff
@@ -195,19 +223,39 @@ public void fullScreenExited() {
...
+                    if (scheme == null || (!scheme.equals("http") && !scheme.equals("https"))) {
+                        // We are ignoring everything that is not http or https. This is a limitation of
+                        // Android's download manager. There's no reason to show a download dialog for
+                        // something we can't download anyways.
+                        Log.w(TAG, "Ignoring download from non http(s) URL: " + response.uri);
+                        return;
+                    }
+
+                    if (callback != null) {
+                        // TODO: get user agent from GeckoView
```
Comment (pocmo) [inline app/src/gecko/java/org/mozilla/focus/web/WebViewProvider.java]: Please file an issue for that and add the issue number here. :)

**Pull request merged**